### PR TITLE
SS-227 Feature/jupyter-lab

### PR DESCRIPTION
**Pull request by alfredeen**
Added jupyter-lab image tagged latest.

Verifications consist of
- pulling the image
- starting a docker container
- basic communication with the container
- adding a simple jupyter notebook
- executing code cells
- verifying the cell output
 
Additional notebook tests can be added in the future. This version is for the dev/test environment. It will be extended with production configuration, Trivy scans, and package creation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,39 @@ All files related to images used in Serve
...
+
+Run docker as a non-root user.
+
+See https://docs.docker.com/engine/install/linux-postinstall/
+
+Do not forget to switch to the docker group in every new terminal.
+
+```
+newgrp docker
+```
```
Comment (sandstromviktor) [inline README.md:41]: Nice update to readme. We should do even more stuff. Explain our method of pulling latest, modifying, testing and publishing.

File: jupyter-lab/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+FROM jupyter/minimal-notebook:latest
+LABEL org.opencontainers.image.source https://github.com/scilifelabdatacentre/stackn-images
+
+USER root
+RUN /bin/bash -c "apt update"
+RUN /bin/bash -c "apt install curl -y"
```
Comment (sandstromviktor) [inline jupyter-lab/Dockerfile:6]: Nice. We can reduce the number of container layers in this file by updating for example:
```code
RUN /bin/bash -c "apt update"
RUN /bin/bash -c "apt install curl -y"
```
to 
```code
RUN /bin/bash -c "apt update && \
    /bin/bash -c "apt install curl -y"
```
and i think we can reduce 
```code
RUN pip uninstall minio -y
RUN pip install minio==7.0.2
```
to
```code
RUN  pip install --force-reinstall "minio==7.0.2"
```

I'll mark this as `review` but change what you believe is necessary.

File: jupyter-lab/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+COPY cli /home/jovyan/cli
+COPY examples /home/jovyan/work/examples
```
Comment (sandstromviktor) [inline jupyter-lab/Dockerfile:8]: Now we need these folders when deploying. I think we should update this so it pulls the folder from the git repo. What do you think?

File: jupyter-lab/tests/test_api.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+)
+
+time.sleep(20)
+
+container.reload()
+
+
+# Tests
+def test_jupyterlab_status_is_running():
+    """Test that the jupyter-lab container is running."""
```
Comment (sandstromviktor) [inline jupyter-lab/tests/test_api.py:41]: Nice that you comment the code like this. 
I think we are using different conventions for commenting. I usually use `# comment`.
We should decide on a convention and stick to that. I'll create an issue regarding this.

**Review by sandstromviktor — COMMENTED** [review]
Great work! Some small changes is needed i think. Let's discuss later today.

**Review by sandstromviktor — COMMENTED** [review]
Great work! Some small changes is needed i think. Let's discuss later today.

File: jupyter-lab/Dockerfile (hunk 2 above)
Comment (alfredeen) [reply jupyter-lab/Dockerfile:6]: Good. I will make the change.

File: README.md (hunk 1 above)
Comment (alfredeen) [reply README.md:41]: Agree

File: jupyter-lab/tests/test_api.py (hunk 4 above)
Comment (alfredeen) [reply jupyter-lab/tests/test_api.py:41]: I use # for comments but docstrings reStructuredText for documenting classes, methods etc.

Comment (alfredeen) [reply jupyter-lab/tests/test_api.py:41]: Agree. We can discuss hot to comment and document the same way.

File: jupyter-lab/Dockerfile (hunk 3 above)
Comment (alfredeen) [reply jupyter-lab/Dockerfile:8]: Yes, agree.

**Pull request merged**